python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -795,35 +823,66 @@ def _plot_wheel_ticks_or_speeds(self, type):
             else:
                 vehicle_time_source = SystemTimeSource.P1_TIME
 
-            figure['layout']['annotations'][0]['text'] += '<br>Vehicle %s Time Source: %s' % \
-                                                          (type.title(),
-                                                           self._time_source_to_display_name(vehicle_time_source))
-
-        if wheel_time_source is not None and vehicle_time_source is not None:
-            if wheel_time_source != vehicle_time_source:
+        same_time_source = True
+        if wheel_time_source is None:
+            common_time_source = vehicle_time_source
+        elif vehicle_time_source is None:
+            common_time_source = wheel_time_source
+        else:
```

Comment:
Will this do what you expect if both time sources are `None`?

Reply:
That should be handled earlier when we load the data. If we don't have either wheel or vehicle measurements, we'll return early and not create the figure.